chapters/annotations.tex

```diff
@@ -888,6 +888,14 @@ \subsection{Component Instance}\label{component-instance}
 Sub-connectors in a hierarchical connector are only shown when they can be connected to.
 Non-connector components are only shown in the diagram layer.
 
+\subsubsection{Default Outline for Missing Graphics}\label{default-outline-for-missing-graphics}
+If the icon of a component placed in a diagram layer does not contain any graphical primitives (including inherited ones, and regardless of \lstinline!visible!-attributes; but excluding connectors), tools shall show a tool-dependent rudimentary outline of the component's transformed \lstinline!extent!.
+
+\begin{nonnormative}
+The reason for making the tool-dependent outline rudimentary is to encourage the model developer to provide a proper icon.
+Regardless of \lstinline!visible!-attributes imply that if the primitives have a parameter-dependent value for \lstinline!visible! or even \lstinline!visible = false! it will not switch to the default icon regardless of the parameter setting.
```

Comment:
It's weird to include the constant `false` case in this sentence (also fix grammar):
```suggestion
That \lstinline!visible!-attributes are not regaded implies that if the primitives have a parameter-dependent value for \lstinline!visible!, the display will not switch to the default icon regardless of the parameter setting.
```

Reply:
I don't want to explicitly add that work-around in the specification, and could just change "parameter-dependent" to "possibly parameter-dependent", and then remove "visible=false" as it is clearly covered by "possibly parameter-dependent".

Reply:
I think that _possibly parameter-dependent_ makes it hard to finish the sentence; _regardless of the possibly parameter-dependent setting_ doesn't read very nice.

To me, the part about an icon only showing connectors doesn't have to be seen as a workaround for a problem with the specification; I prefer to see it as a useful application of the rule about non-visible primitives.  It's not a big deal what we say here, but with my concerned colleague in mind I would prefer to at least have a poll to refer to if we end up not explaining this use case.

Reply:
I have now shortened it as much as possible; I realized that since we don't care for the value of the attribute we don't have to consider whether it is parameter-dependent or not.